### PR TITLE
Core/Movement: Fix warrior charge animation

### PR DESCRIPTION
**Pull request by LichKing255**
##### Current behaviour:

Warrior appears like teleporting, the charge animation starts, then warrior moves about 2-3 yards forward, then gets back, and immediately teleported to the target after that.

##### Expected behaviour:

Charge animation should be smooth.

##### HOW TO TEST THE CHANGES:

1. Make Warrior
2. Make another character as target
3. Jump backwards and immediately press charge

Charge animation always looks smooth on the warrior's client, but very weird and broken on the target's client (this is not always happening, this macro to reproduce it easily:

> /cast charge
> /target player
> .cool
> /targetlasttarget

### CODE REVIEW

File: src/server/game/Handlers/MovementHandler.cpp [hunk 1]
```diff
@@ -322,21 +322,27 @@ void WorldSession::HandleMovementOpcodes(WorldPacket & recvData)
...
         recvData.rfinish();                     // prevent warnings spam
         return;
     }
 
-    recvData.rfinish();                         // prevent warnings spam
+    if (!mover->movespline->Finalized())
+    {
+        recvData.rfinish();                     // prevent warnings spam
+        return;
+    }
```
Comment (Helias) [inline src/server/game/Handlers/MovementHandler.cpp]: This check is already contained in this PR https://github.com/azerothcore/azerothcore-wotlk/pull/2410

**Comment by Winfidonarleyan** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/2410/files#diff-649b0ea06820eb06283d1ccc8c83efdbR329

**Comment by pklloveyou** [comment]
PR #2410　It should have fixed this bug　。I had another client that did not appear to intercept the Caton error after merging#2410

**Comment by pklloveyou** [comment]
Is there a way to replicate this error after today's latest commit?I'll test it.

**Pull request closed without merging**